kafka-client/src/main/java/dev/responsive/kafka/internal/db/SegmentableKey.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db;
+
+public interface SegmentableKey<T> extends Comparable<T> {
+  /**
+   * @return the timestamp that partitioners can use to partition
+   *         keys of this type into different segments.
+   */
+  long segmentTimestamp();
```

Comment:
nit: this is only used when passed into the `segmentId()` function. I think this class would make more sense if this just returned the `segmentId` directly instead of exposing a segment timestamp.

Reply:
The segment id returned is based on other things such as segment interval, which is out of scope / responsibility for these keys. I considered having the `segmentIntervalMs` parameter passed into this function to return the actual `segmentId` but thought it felt worse than just returning a timestamp according to which the key-space is segmentable.

Additionally the timestamp => segment ID calculation is also done in the partitioner (to figure out which segments it should search, for instance, given a raw timestamp), which would mean we do this math in two separate places:
```    
return Long.max(0, windowTimestamp / segmentIntervalMs);
```
I'm open to making that change, though if you feel strongly about it.

Reply:
FWIW I agree with Antoine here -- best to keep this a "pure" key class that only contains data about itself, and let the partitioner determine how that key data maps to a specific segment/partition/etc